mlflow/pyfunc/__init__.py

```diff
@@ -321,17 +322,47 @@ def _enforce_type(name, values: pandas.Series, t: DataType):
         return values
 
     numpy_type = t.to_numpy()
-    is_compatible_type = values.dtype.kind == numpy_type.kind
-    is_upcast = values.dtype.itemsize <= numpy_type.itemsize
-    if is_compatible_type and is_upcast:
+    if values.dtype.kind == numpy_type.kind:
+        is_upcast = values.dtype.itemsize <= numpy_type.itemsize
+    elif values.dtype.kind == "u" and numpy_type.kind == "i":
+        is_upcast = values.dtype.itemsize < numpy_type.itemsize
+    elif values.dtype.kind in ("i", "u") and numpy_type == np.float64:
+        # allow (u)int => double conversion
+        is_upcast = values.dtype.itemsize <= 6
```

Comment:
Why is `6` the threshold here? I know we need < 8. Is 7 bad (does a 7 byte dtype even exist?). Maybe an explanatory comment would help.

Reply:
Double has 52 bits of mantissa, so the size has to be < 52. 6 is the smallest multiple of 8 smaller than 52. I guess we don't have to be that specific and just say integers or smaller.